tests/sklearn/test_sklearn_models.py

```diff
@@ -1120,6 +1137,212 @@ def check_exposition_structural_methods_decision_trees(model, x, y):
     )
 
 
+# Add a test to match fp32 -> quant -> dequant weights at the ONNX level
+# in the high bit width setting
+# Some snippet to do this:
+# if n_bits > 17:
+#     diff = init_tensor - init_tensor_as_int
+#     max_diff = numpy.abs(diff).max()
+#     if max_diff > 1e-3:
+#         raise ValueError(f"{max_diff=} > 1e-4")
+
+
+# pylint: disable-next=too-many-locals,too-many-statements
+# TODO: make this pass with rounding (high-bit-width create overflow)
+@pytest.mark.parametrize("model_class, parameters", get_sklearn_tree_models_and_datasets())
+@pytest.mark.parametrize("use_rounding", [False, True])
+def test_load_fitted_sklearn_tree_models(
+    subtests,
+    model_class,
+    parameters,
+    use_rounding,
+    load_data,
+    is_weekly_option,
+    verbose=True,
+):
+    """Test `from_sklearn_model` functionnality of tree-based models."""
+
+    numpy.random.seed(0)
+    os.environ["TREES_USE_ROUNDING"] = str(int(use_rounding))
+
+    x, y = get_dataset(
+        model_class, parameters, min(N_BITS_REGULAR_BUILDS), load_data, is_weekly_option
+    )
+
+    if verbose:
+        print("Run check_load_pre_trained_sklearn_models")
+
+    assert issubclass(model_class, BaseTreeEstimatorMixin)
+    concrete_model = instantiate_model_generic(model_class, n_bits=min(N_BITS_REGULAR_BUILDS))
+    # Fit the model and retrieve both the Concrete ML and the scikit-learn models
+    with warnings.catch_warnings():
+        # Sometimes, we miss convergence, which is not a problem for our test
+        warnings.simplefilter("ignore", category=ConvergenceWarning)
+        concrete_model, sklearn_model = concrete_model.fit_benchmark(x, y)
+
+    # This step is needed in order to handle partial classes
+    model_class = get_model_class(model_class)
+    max_n_bits = 18
+    reasonable_n_bits = 8
+
+    # TODO: add normal bit-width comparison
+    if isinstance(concrete_model, BaseTreeClassifierMixin):
```

Comment:
if you are checking if this model is. a classifier, better use our `is_classifier_or_partial_classifier` utility tool !

Reply:
Doesn't look clearer to me tbh

But I can change it if you want.

Reply:
not clearer indeed, but safer / more robust imo

Reply:
changed